libs/math/include/math/quat.h

```diff
@@ -98,8 +98,7 @@ class MATH_EMPTY_BASES TQuaternion :
 
     // handles implicit conversion to a tvec4. must not be explicit.
     template<typename A, typename = enable_if_arithmetic_t<T>>
-    constexpr TQuaternion(A w) : x(0), y(0), z(0), w(w) {
-    }
+    constexpr explicit TQuaternion(A w) : x(0), y(0), z(0), w(w) {}
```

Comment:
why explicit here?

Reply:
I think the issue is that when we have implicit conversions from float to tvec3, tvec4, and quat, 
and also `TVec3::operator*()` which accept tvec3, tvec4, quat, the compiler doesn't know what function to use when it sees: `tvec3 * float`.

Let me try to unset this so I can show the errors I get with msvc.

Reply:
Just making this one  constructor not explicit, and building Scene.cpp, I have all the following errors. I don't think we rely on this constructor not being explicit anywhere in the code, and it seems clearer to me that float cannot convert transparently to quaternions, no? Also is a quaternion initialized to (0, 0, 0, w) useful at all?

```
1>------ Build started: Project: filament, Configuration: Debug x64 ------
1>Scene.cpp
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3385,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +': 4 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(109,18): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +(size_t)'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(108,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<__int64> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<__int64> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<__int64> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3385,1): message :  while trying to match the argument list '(_RanIt, utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3385,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3385,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3385,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3385,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3454): message :  see reference to function template instantiation 'std::pair<_RanIt,_RanIt> std::_Partition_by_median_guess_unchecked<_RanIt,_Pr>(_RanIt,_RanIt,_Pr)' being compiled
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3454): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3454): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3454): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>,
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3454): message :             _Pr=filament::details::FScene::prepareDynamicLights::<lambda_1>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3454): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3480): message :  see reference to function template instantiation 'void std::_Sort_unchecked<_Iter,_Fn>(_RanIt,_RanIt,__int64,_Pr)' being compiled
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3480): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3480): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3480): message :             _Iter=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>,
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3480): message :             _Fn=filament::details::FScene::prepareDynamicLights::<lambda_1>,
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3480): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>,
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3480): message :             _Pr=filament::details::FScene::prepareDynamicLights::<lambda_1>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3480): message :         ]
1>C:\greg\github\filament\filament\src\Scene.cpp(257): message :  see reference to function template instantiation 'void std::sort<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>,filament::details::FScene::prepareDynamicLights::<lambda_1>>(const _RanIt,const _RanIt,_Pr)' being compiled
1>C:\greg\github\filament\filament\src\Scene.cpp(257): message :         with
1>C:\greg\github\filament\filament\src\Scene.cpp(257): message :         [
1>C:\greg\github\filament\filament\src\Scene.cpp(257): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>,
1>C:\greg\github\filament\filament\src\Scene.cpp(257): message :             _Pr=filament::details::FScene::prepareDynamicLights::<lambda_1>
1>C:\greg\github\filament\filament\src\Scene.cpp(257): message :         ]
1>C:\greg\github\filament\filament\include\filament/Color.h(142): message :  see reference to class template instantiation 'filament::math::details::TVec4<float>' being compiled
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3386,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -': 3 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(112,21): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(const utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> &) const'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(110,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<uint32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3386,1): message :  while trying to match the argument list '(_RanIt, int)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3386,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3386,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3386,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3386,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3386,5): error C2672:  '_Guess_median_unchecked': no matching overloaded function found
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3386,1): error C2780:  'void std::_Guess_median_unchecked(_RanIt,_RanIt,_RanIt,_Pr)': expects 4 arguments - 3 provided
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3366): message :  see declaration of 'std::_Guess_median_unchecked'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3388,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +': 4 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(109,18): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +(size_t)'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(108,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator +<int,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator +<int,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator +<int,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator +<int,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator +<int,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator +<int,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator +<int,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator +<int,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<int32_t> filament::math::details::operator +<int,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<int32_t> filament::math::details::operator +<int,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator +<int,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator +<int,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<uint32_t> filament::math::details::operator +<int,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator +<int,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3388,1): message :  while trying to match the argument list '(_RanIt, int)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3388,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3388,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3388,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3388,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -': 3 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(112,21): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(const utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> &) const'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(110,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<uint32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,1): message :  while trying to match the argument list '(_RanIt, int)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,33): error C2672:  '_Debug_lt_pred': no matching overloaded function found
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,1): error C2780:  'bool std::_Debug_lt_pred(_Pr &&,_Ty1 &&,_Ty2 &&) noexcept(<expr>)': expects 3 arguments - 2 provided
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\xutility(632): message :  see declaration of 'std::_Debug_lt_pred'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,111): error C2088:  '-': illegal for class
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,85): error C2672:  'operator __surrogate_func': no matching overloaded function found
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3390,1): error C2780:  'auto filament::details::FScene::prepareDynamicLights::<lambda_1>::operator ()(const _T1 &,const _T2 &) const': expects 2 arguments - 1 provided
1>C:\greg\github\filament\filament\src\Scene.cpp(257): message :  see declaration of 'filament::details::FScene::prepareDynamicLights::<lambda_1>::operator ()'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3415,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -': 3 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(112,21): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(const utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> &) const'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(110,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<uint32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3415,1): message :  while trying to match the argument list '(_RanIt, int)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3415,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3415,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3415,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3415,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3415,17): error C2672:  '_Debug_lt_pred': no matching overloaded function found
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3415,1): error C2780:  'bool std::_Debug_lt_pred(_Pr &&,_Ty1 &&,_Ty2 &&) noexcept(<expr>)': expects 3 arguments - 2 provided
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\xutility(632): message :  see declaration of 'std::_Debug_lt_pred'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3416,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -': 3 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(112,21): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(const utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> &) const'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(110,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<uint32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3416,1): message :  while trying to match the argument list '(_RanIt, int)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3416,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3416,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3416,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3416,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3416,24): error C2672:  'operator __surrogate_func': no matching overloaded function found
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3416,1): error C2780:  'auto filament::details::FScene::prepareDynamicLights::<lambda_1>::operator ()(const _T1 &,const _T2 &) const': expects 2 arguments - 1 provided
1>C:\greg\github\filament\filament\src\Scene.cpp(257): message :  see declaration of 'filament::details::FScene::prepareDynamicLights::<lambda_1>::operator ()'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3418,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -': 3 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(112,21): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(const utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> &) const'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(110,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<uint32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3418,1): message :  while trying to match the argument list '(_RanIt, int)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3418,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3418,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3418,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3418,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3419,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -': 3 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(112,21): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(const utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> &) const'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(110,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator -(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec3<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<int32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator -<int,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TVec2<uint32_t> filament::math::details::operator -<int,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator -<int,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :             U=int
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(142,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3419,1): message :  while trying to match the argument list '(_RanIt, int)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3419,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3419,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3419,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3419,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3419,22): error C2672:  'std::iter_swap': no matching overloaded function found
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3419,1): error C2780:  'void std::iter_swap(_FwdIt1,_FwdIt2)': expects 2 arguments - 1 provided
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\utility(33): message :  see declaration of 'std::iter_swap'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): error C2666:  'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +': 4 overloads have similar conversions
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(109,18): message :  could be 'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +(size_t)'
1>C:\greg\github\filament\libs\utils\include\utils/Zip2Iterator.h(108,18): message :  or       'utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *> utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::operator +(size_t) const'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec3<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec2<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec4<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec4<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TQuaternion<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TQuaternion<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec4<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec4<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec2<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec3<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat33<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TMat33<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec3<__int64> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec3<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<__int64> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec2<bool> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TQuaternion<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TQuaternion<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat44<float> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TMat44<float> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TVec2<__int64> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TVec2<uint32_t> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :  or       'filament::math::details::TMat44<double> filament::math::details::operator +<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type,void>(const filament::math::details::TMat44<double> &,U)'
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         with
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         [
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :             U=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>::difference_type
1>C:\greg\github\filament\libs\math\include\math/TVecHelpers.h(128,14): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): message :  while trying to match the argument list '(_RanIt, __int64)'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3468): message :  see reference to function template instantiation 'void std::_Make_heap_unchecked<_RanIt,_Pr>(_RanIt,_RanIt,_Pr)' being compiled
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3468): message :         with
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3468): message :         [
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3468): message :             _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2> *>,float *>,
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3468): message :             _Pr=filament::details::FScene::prepareDynamicLights::<lambda_1>
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(3468): message :         ]
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,43): error C2672:  'std::move': no matching overloaded function found
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): error C2780:  '_DestTy *std::move(_ExPo &&,_FwdIt1,_FwdIt1,_DestTy (&)[_DestSize]) noexcept': expects 4 arguments - 1 provided
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\xutility(2131): message :  see declaration of 'std::move'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): error C2780:  '_FwdIt2 std::move(_ExPo &&,_FwdIt1,_FwdIt1,_FwdIt2) noexcept': expects 4 arguments - 1 provided
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\xutility(2121): message :  see declaration of 'std::move'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): error C2780:  '_DestTy *std::move(_InIt,_InIt,_DestTy (&)[_DestSize])': expects 3 arguments - 1 provided
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\xutility(2114): message :  see declaration of 'std::move'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726,1): error C2780:  '_OutIt std::move(_InIt,_InIt,_OutIt)': expects 3 arguments - 1 provided
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\xutility(2102): message :  see declaration of 'std::move'
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.22.27905\include\algorithm(2726): error C2512:  'std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::details::FLightManager::Instance,filament::details::Culler::result_type,filament::math::float2>::Structure,float>::pair': no appropriate default constructor available
1>Done building project "filament.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

Reply:
This is fixed with #1695